### PR TITLE
Feature/store general-miner-info

### PR DESCRIPTION
**Pull request by nmlinaric**
Created endpoints:
- ```/user/node/generalminerinfo``` which uses PUT for storing/updating node general info
- ```/user/node/generalminerinfo/:nodeId``` for fetching node general info by node id

### CODE REVIEW

File: src/Controller/Api/NodeGeneralInfoController.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    public async updateOrCreateNodeGenralInfo(
+        req: ValidatedRequest<CreateNodeGeneralInfoRequestSchema>,
+        res: Response) {
+        try {
+            const {version, sectorSize, minerPower, totalPower} = req.body;
+            const nodeId = res.locals.node.id;
+            const result = await this.nodeGeneralInfoService.updateOrCreateNodeGenralInfo(
+                version, sectorSize, minerPower, totalPower, nodeId);
+            if (result) {
```
Comment (morrigan) [inline src/Controller/Api/NodeGeneralInfoController.ts]: Can this be null? If yes, this request will end up hanging.
If not, you can write without if and just return result.

File: src/Controller/Api/NodeGeneralInfoController.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    public async fetchNodeGenralInfo(req: Request, res: Response) {
+        try {
+            const nodeId = res.locals.node.id;
+            const result = await this.nodeGeneralInfoService.fetchNodeGenralInfo(nodeId);
+            if (result) {
+                res.status(200).json(result)
+            }
+        } catch (e) {
+            console.log(e)
```
Comment (morrigan) [inline src/Controller/Api/NodeGeneralInfoController.ts]: remove console log (if you want, you can add this rule in tslint for warning/error)

File: src/Controller/Api/NodeGeneralInfoController.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            logger.error(`Error occurred on storing general info in controller: ${e.message}`);
+            res.status(500).json({error: "An unknown error occurred."});
+        }
+    }
+
+    public async fetchNodeGenralInfo(req: Request, res: Response) {
+        try {
+            const nodeId = res.locals.node.id;
+            const result = await this.nodeGeneralInfoService.fetchNodeGenralInfo(nodeId);
+            if (result) {
```
Comment (morrigan) [inline src/Controller/Api/NodeGeneralInfoController.ts]: Same thinking here - make sure how your code works.

File: src/Services/NodeGeneralInfoService.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import {NodeGeneralInfo} from "../Models/NodeGeneralInfo";
+
+export class NodeGeneralInfoService {
+
+    public async updateOrCreateNodeGenralInfo(
+        version: string, sectorSize: number, minerPower: number, totalPower: number, nodeId: number) {
+        const node = await NodeGeneralInfo.findOne({
```
Comment (morrigan) [inline src/Services/NodeGeneralInfoService.ts]: Why don't you use the below method `fetchNodeGenralInfo` rather than duplicate code?

Comment (nmlinaric) [reply src/Services/NodeGeneralInfoService.ts]: Old habits, still in the process of reverting to DRY.

**Pull request merged**